### PR TITLE
Dummy project and modified release process

### PR DESCRIPTION
**Pull request by lassemand**
## Purpose

Initial commit to include new project

## Changes

Add dummy project with corresponding CI
Modify release process to prepare for generalisation
closes https://github.com/Concordium/concordium-misc-tools/issues/150

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -26,16 +30,15 @@ jobs:
     - name: Extract Cargo package version
       id: cargo_version
       run: |
-        cd recover-id-object
+        cd ${{ matrix.service_name }}
         VERSION=$(cargo metadata --no-deps --format-version 1 | jq -r '.packages[0].version')
-        echo "Extracted version is $VERSION"
-        echo "::set-output name=version::$VERSION"
```
Comment (lassemand) [inline .github/workflows/release.yml:32]: This is no longer the recommended way.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -11,6 +11,10 @@ jobs:
   build-and-push:
     runs-on: ubuntu-latest
     environment: release-recover-id-object
+    strategy:
+      matrix:
+        service_name:
+          - recover-id-object
```
Comment (lassemand) [inline .github/workflows/release.yml:17]: We might as well reuse the definition

File: notification-server/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[package]
+name = "notification-server"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+
+concordium-rust-sdk = { version = "*", path = "../deps/concordium-rust-sdk" }
```
Comment (lassemand) [inline notification-server/Cargo.toml:10]: Basically just added to ensure local dev is setup correctly

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -11,6 +11,10 @@ jobs:
   build-and-push:
     runs-on: ubuntu-latest
     environment: release-recover-id-object
```
Comment (DOBEN) [inline .github/workflows/release.yml:13]: The environment should be renamed if we intend to use this release with a matrix and several services.

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -33,6 +33,7 @@ jobs:
           - recover-id-object/Cargo.toml
           - generator/Cargo.toml
           - wallet-connect-test-bench/smart-contract/Cargo.toml
+          - notification-server/Cargo.toml
```
Comment (DOBEN) [inline .github/workflows/ci.yaml:36]: The related issue will be closed automatically by this PR at merge when you add "closes #150" in the PR description which I recommend adding.

File: notification-server/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+[package]
+name = "notification-server"
+version = "0.1.0"
+edition = "2021"
```
Comment (DOBEN) [inline notification-server/Cargo.toml:4]: I suggest to add the author, so people (even after deleting the git history) have another way to figure out the origin/author of the code.

authors = ["Concordium AG <developers@concordium.com>"]

**Review by DOBEN — APPROVED** [review]
LGTM

File: .github/workflows/release.yml (hunk 4 above)
Comment (lassemand) [reply .github/workflows/release.yml:13]: Will do that in a new PR when I actually am ready to release. :)

Comment (lassemand) [reply .github/workflows/release.yml:13]: I will remove matrix changes until it may be an issue.

**Pull request merged**